src/main/java/com/palantir/gradle/gitversion/GitVersionPlugin.java

```diff
@@ -17,37 +17,32 @@
 package com.palantir.gradle.gitversion;
 
 import groovy.lang.Closure;
-import java.io.File;
-import java.io.IOException;
-import org.eclipse.jgit.api.Git;
-import org.eclipse.jgit.internal.storage.file.FileRepository;
 import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
+import org.gradle.api.provider.Provider;
 
 public final class GitVersionPlugin implements Plugin<Project> {
-    private final Timer timer = new Timer();
 
     @Override
     public void apply(final Project project) {
         project.getRootProject().getPluginManager().apply(GitVersionRootPlugin.class);
 
-        final Git git = gitRepo(project);
+        Provider<GitVersionCacheService> serviceProvider = project.getGradle()
+                .getSharedServices()
+                .registerIfAbsent("GitVersionCacheService", GitVersionCacheService.class, _spec -> {});
```

Comment:
I would make a static method to get the `Provider<GitVersionCacheService>` from a `Project` which is shared by both plugins.

Reply:
You could even put it on `GitVersionCacheService`